Content/Items/Jungle/Weapons.HauntedDaggerStaff.cs

```diff
@@ -0,0 +1,509 @@
+
+			Color color = new Color(70, 200, 100, 0) * 0.4f * Utils.Clamp((float)Math.Sin(Main.GlobalTimeWrappedHourly * 2f), 0.5f, 1f);
+
+			effect.Parameters["uColor"].SetValue(color.ToVector4());
+			effect.Parameters["noiseImage1"].SetValue(ModContent.Request<Texture2D>(AssetDirectory.Assets + "Noise/PerlinNoise").Value);
+
+			effect.CurrentTechnique.Passes[0].Apply();
+
+			Main.spriteBatch.Draw(texGlow, Projectile.Center - Main.screenPosition, null, new Color(70, 200, 100, 0), Projectile.rotation + MathHelper.ToRadians(rotTimer), texGlow.Size() / 2f, Projectile.scale * 1.5f, 0f, 0f);
+
+			Main.spriteBatch.Draw(bloomTex, Projectile.Center - Main.screenPosition, null, new Color(70, 200, 100, 0), Projectile.rotation + MathHelper.ToRadians(rotTimer), bloomTex.Size() / 2f, 1f, 0f, 0f);
+
+			Main.spriteBatch.End();
+			Main.spriteBatch.Begin(default, default, default, default, default, default, Main.GameViewMatrix.TransformationMatrix);
+
+			return false;
+		}
+
+		public void Unembed(bool dealDamage, NPC target)
+		{
+			var gNPC = target.GetGlobalNPC<HauntedDaggerGlobalNPC>();
+
+			gNPC.Unembed = dealDamage;
+			gNPC.UnembeddingPlayerWhoAmI = Owner.whoAmI;
+
+			int count = gNPC.GetDaggerCount(target);
+
+			if (gNPC.UnembedCount <= 0)
+				gNPC.UnembedCount = count;
+
+			Projectile.ai[0] = 0f;
+			TargetWhoAmI = -1f;
+			AttackDelay = maxAttackDelay;
+
+			Projectile.velocity *= -1f;
+			attackTimer = 0;
+
+			for (int i = 0; i < 6; i++)
+			{
+				Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(2f, 2f), ModContent.DustType<Dusts.GlowFastDecelerate>(),
+					Projectile.Center.DirectionTo(Projectile.Center + Projectile.velocity).RotatedByRandom(1f) * Main.rand.NextFloat(16f), 0, new Color(70, 200, 100), 0.5f);
+			}
+		}
+
+		internal void DoIdleMovement()
+		{
+			float totalCount = Owner.ownedProjectileCounts[Type] > 0 ? (float)Owner.ownedProjectileCounts[Type] : 1;
+
+			var idlePos = Owner.Center + new Vector2(0f, -100) + new Vector2(0, 35).RotatedBy(MathHelper.ToRadians(Projectile.minionPos / totalCount * 360f + (Projectile.minionPos == totalCount ? 360f / totalCount : 0f)) + MathHelper.ToRadians(Lifetime));
+
+			float dist = Vector2.Distance(Projectile.Center, idlePos);
+
+			Vector2 toIdlePos = idlePos - Projectile.Center;
+			if (toIdlePos.Length() < 0.0001f)
+			{
+				toIdlePos = Vector2.Zero;
+			}
+			else
+			{
+				float speed = 35f;
+				if (dist < 1000f)
+					speed = MathHelper.Lerp(5f, 35f, dist / 1000f);
+
+				if (dist < 100f)
+					speed = MathHelper.Lerp(0.1f, 5f, dist / 100f);
+
+				toIdlePos.Normalize();
+				toIdlePos *= speed;
+			}
+
+			Projectile.velocity = (Projectile.velocity * (25f - 1) + toIdlePos) / 25f;
+
+			if (dist > 2000f)
+			{
+				Projectile.Center = idlePos;
+				Projectile.velocity = Vector2.Zero;
+				Projectile.netUpdate = true;
+			}
+
+			Projectile.rotation = Projectile.velocity.X * 0.05f;
+		}
+
+		internal void UpdateProjectile()
```

Comment:
rename this "UpdateProjectileLifetime" or something in that vein

Reply:
fixed